File: fixtures/python/bind_all_interfaces/positive.py
# SEC013-bind-all-interfaces: positive fixture
# Demonstrates wide-open server bind addresses that should be flagged.

import socket
import flask
import uvicorn

# FLAGGED: Flask app.run with IPv4 any-address.
app = flask.Flask(__name__)
app.run("0.0.0.0", port=5000)

# FLAGGED: uvicorn.run with host="0.0.0.0" keyword argument.
uvicorn.run(app, host="0.0.0.0", port=8000)

# FLAGGED: socket.bind with tuple (any-address, port).
s = socket.socket()
s.bind(("0.0.0.0", 8080))

# FLAGGED: bare bind with IPv6 any-address.
bind("::")

File: fixtures/js/bind_all_interfaces/positive.ts
// SEC013-bind-all-interfaces: positive fixture
// Demonstrates wide-open server bind addresses that should be flagged.

import http from 'http';
import express from 'express';

const app = express();

// FLAGGED: app.listen with IPv4 any-address.
app.listen("0.0.0.0", 3000);

// FLAGGED: server.listen with host:port string form.
const server = http.createServer(app);
server.listen("0.0.0.0:8080");

// FLAGGED: socket.bind with IPv6 any-address.
const sock = {} as any;
sock.bind("::");
